lib/config.js

```diff
@@ -229,6 +230,16 @@ function getLocalConfig(thisConfig, directory) {
             continue;
         }
 
+        // Check for root flag
+        if (localConfig.root === true) {
+            rootPath = path.dirname(localConfigFile);
```

Comment:
That was my first thought, but I don't think it works, because there might be a `package.json` with ESLint configuration after an `.eslintrc` file with `root: true`.   Additionally,  I was also not certain I could rely on the order of elements in the `localConfigFiles` array.